Hahnium.Chip8/Chip8Ppu.cs

```diff
@@ -81,25 +82,54 @@ internal void Cycle()
 {RenderRegister(3)}   {RenderRegister(7)}   {RenderRegister(11)}   {RenderRegister(15)}
 FPS {++cycleCount / timer.Elapsed.TotalSeconds}");
 
-            if(timer.Elapsed.TotalSeconds > 1.0)
+            if (timer.Elapsed.TotalSeconds > 1.0)
             {
                 timer = Stopwatch.StartNew();
                 cycleCount = 0;
             }
         }
 
-        private string RenderRegister(int register) => $"V{register:X} ${this.cpu.Registers[register]:X2}";
+        public bool Draw(byte xPosition, byte yPosition, Memory<byte> spriteMemory)
+        {
+            bool bitFlipped = false;
+            using var spriteHandle = spriteMemory.Pin();
+            var sprite = (byte*)spriteHandle.Pointer;
+
+            for (int y = yPosition; y < yPosition + spriteMemory.Length; y++)
+            {
+                byte spriteLine = *sprite;
+
+                for (int x = xPosition; x < xPosition + 8; x++)
+                {
+                    byte pixel = frameBuffer[(y * DisplayWidth) + x];
+                    var spritePixel = (spriteLine & 0b1000_0000) >> 7;
```

Comment:
Maybe remove binary literals to constants to make it more clear what's going on?

Reply:
Personally I think it's clearer like this. I want to grab the most significant bit and shift it down to a 0/1 value. Would it be clearer for everyone to rename it MSB (most significant bit)?

Reply:
Yeah I think so.